realm-jni/src/util.hpp

```diff
@@ -100,6 +100,7 @@ std::string num_to_string(T pNumber)
 #define Q(x)    reinterpret_cast<realm::Query*>(x)
 #define G(x)    reinterpret_cast<realm::Group*>(x)
 #define ROW(x)  reinterpret_cast<realm::Row*>(x)
+#define SG(ptr) reinterpret_cast<SharedGroup*>(ptr)
```

Comment:
You should probably remove the same macro in `io_realm_internal_SharedGroup.cp`.